src/DI/Container.php

```diff
@@ -98,8 +98,10 @@ public function removeService($name)
 	 */
 	public function getService($name)
 	{
-		$name = isset($this->meta[self::ALIASES][$name]) ? $this->meta[self::ALIASES][$name] : $name;
 		if (!isset($this->registry[$name])) {
+			if (isset($this->meta[self::ALIASES][$name])) {
+				return $this->getService($this->meta[self::ALIASES][$name]);
+			}
```

Comment:
The method call could be inlined, do you like it better?

~~~php
public function getService($name)
{
	if (!isset($this->registry[$name])) {
		if (isset($this->meta[self::ALIASES][$name])) {
			$name = $this->meta[self::ALIASES][$name];
			if (!isset($this->registry[$name])) {
				$this->registry[$name] = $this->createService($name);
			}
		} else {
			$this->registry[$name] = $this->createService($name);
		}
	}
	return $this->registry[$name];
}
~~~